### PR TITLE
[trainer] refactor: code refactor for diffusion training

### PR DESCRIPTION
**Pull request by zhtmike**
### What does this PR do?

Major change:
- Decouple LLM and diffusion configs, losses, etc., for easier integration of diffusion models/algorithms/engines in the future. (resolve comment from #5951)
  - Diffusion configs are now all placed in `verl/trainer/config/diffusion`, including `diffusion_actor`, `dp_diffusion_actor`, `diffusion_fsdp`, `diffusion_rollout`, etc.
  - Diffusion config classes are now all placed in `verl/workers/config/diffusion`, including `actor`, `model`, `rollout`.
  - Clean LLM-specific configs such as `use_remove_padding`, or unused configs. The content of `_generated_diffusion_trainer.yaml` has now been reduced by ~60%.

Other changes:
- Clean the diffusion agent loop output; remove dead code originally borrowed from the LLM agent loop, such as `input_id`, `attention_mask`, etc., which are not used in diffusion training currently.
- Refactor and remove `extra_configs` in diffusion configs. It is too loose and may cause confusion for users. (resolve comment from #5951)
- Decouple LLM and diffusion losses to make diffusion losses clearer.
- Code cleanup for diffusion trainer, diffusion agent loop, etc.
- Change the vllm-omni rollout import to registration, following the diffusion trainer pattern. (resolve comment from #5951)

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `vllm_omni`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: verl/experimental/agent_loop/diffusion_agent_loop.py [hunk 1]
```diff
@@ -35,9 +36,16 @@
...
 from verl.utils.config import omega_conf_to_dataclass
 from verl.utils.dataset.rl_dataset import get_dataset_class
+from verl.utils.profiler import simple_timer
 from verl.workers.config import DiffusionModelConfig, DiffusionRolloutConfig
 
 
+def _config_to_sampling_dict(config: Optional[BaseConfig]) -> dict:
+    if config is None:
+        return {}
+    return {k: v for k, v in config.items() if not k.startswith("_")}
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply verl/experimental/agent_loop/diffusion_agent_loop.py:46]: no need

File: verl/experimental/agent_loop/diffusion_agent_loop.py [hunk 2]
```diff
@@ -226,73 +230,68 @@ async def _agent_loop_postprocess(self, output, **kwargs) -> _InternalDiffusionA
...
+
+                data = DataProto(
+                    batch=batch,
+                    non_tensor_batch=non_tensor_batch,
+                )
+                selected_reward_loop_worker_handle = random.choice(self.reward_loop_worker_handles)
+                result = await selected_reward_loop_worker_handle.compute_score.remote(data)
+                output.reward_score = result["reward_score"]
+                output.extra_fields["reward_extra_info"] = result["reward_extra_info"]
+            output.metrics.compute_score = timing["compute_score"]
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply verl/experimental/agent_loop/diffusion_agent_loop.py:294]: no need

**Comment by zhtmike** [comment]
@knlnguyen1802 please take a look of vllm-omni change thx, I think it is better to load the custom class directly in vllm-omni side?
